packages/react-menu/src/components/MenuTrigger/useMenuTrigger.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import { MenuTriggerProps, MenuTriggerState } from './MenuTrigger.types';
+import { useMenuContext, MenuContextValue } from '../../menuContext';
+
+export const menuTriggerShorthandProps: (keyof MenuTriggerProps)[] = [];
+
+const mergeProps = makeMergeProps<MenuTriggerState>({ deepMerge: menuTriggerShorthandProps });
+
+/**
+ * Create the state required to render MenuTrigger.
+ * Clones the only child component and adds necessary event handling behaviours to open a popup menu
+ *
+ * @param props - props from this instance of MenuTrigger
+ * @param ref - reference to root HTMLElement of MenuTrigger
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ *
+ * {@docCategory MenuTrigger }
+ */
+export const useMenuTrigger = (
+  props: MenuTriggerProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: MenuTriggerProps,
+): MenuTriggerState => {
+  const setOpen = useMenuContext(context => context.setOpen);
+
+  const state = mergeProps(
+    {
+      ref: useMergedRefs(ref, React.useRef(null)),
+    },
+    defaultProps,
+    resolveShorthandProps(props, menuTriggerShorthandProps),
+  );
+
+  state.setOpen = setOpen;
+
+  const child = React.Children.only(state.children);
+  state.children = React.cloneElement(child as React.ReactElement, getTriggerProps(state.setOpen));
+
+  return state;
+};
+
+// TODO this is quick 'n dirty, follow up and improve interactions
+const getTriggerProps = (setOpen: MenuContextValue['setOpen']) => {
+  const triggerProps: React.HTMLAttributes<HTMLElement> = {};
+  triggerProps.onClick = e => {
+    setOpen(s => !s);
+  };
```

Comment:
```suggestion

  triggerProps.onClick = useEventCallback((e => {
    setOpen(s => !s);
    triggerProps.onClick(e)
  });
```

_tldr: feel free to ignore for this PR_

---

There are two important things there:
- callbacks should be stable as a child of `MenuTrigger` can be wrapped with `React.memo()` (it was a case in N* multiple times), otherwise we will pass unstable props and it breaks
- we should also call user's handlers to handle a case below (as via `.cloneElement()` we will override existing props)

```tsx
<MenuTrigger>
  <button onClick={() => {}} /> // 💣 this should be also called
</MenuTrigger>
```

Reply:
yeah it's not overlooked, interactions can be messy since so many different ones have to be drilled down from the original child.... as I said in the comment above:

`// TODO this is quick 'n dirty, follow up and improve interactions`